### PR TITLE
Adds linen_exampels/cloud/

### PR DESCRIPTION
**Pull request by andsteing**
This new directory contains scripts for launching examples "fire and forget" style on GCS. Verified with `MNIST` and `imagenet` examples so far.

See #707

### CODE REVIEW

File: linen_examples/cloud/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+five minutes.
+
+## Preparation
+
+Prerequisites:
+
+1. Create a Google Cloud account.
+2. Set up billing : https://console.cloud.google.com/billing
+3. Create a storage bucket (GCS).
+4. Optional : Get quota for accelerators. This is usually granted with a short
```
Comment (avital) [inline linen_examples/cloud/README.md:30]: Please don't put a space before ":"s.

**Review by avital — COMMENTED** [review]
I only skimmed (so far) -- super cool!!

Does this also support using TPUs on Cloud?

File: linen_examples/cloud/README.md (hunk 1 above)
Comment (andsteing) [reply linen_examples/cloud/README.md]: Removed.

**Comment by andsteing** [comment]
> I only skimmed (so far) -- super cool!!
> 
> Does this also support using TPUs on Cloud?

I haven't tried yet, will do when I'll add the `wmt` example (planned next).

**Comment by avital** [comment]
@jheek do you have time for a proper review on this PR?

File: linen_examples/cloud/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+- `$PROJECT`: The name of your Google Cloud project.
+- `$GCS_BUCKET`: The name of the Google Cloud Storage bucket where the model
+  output (artifacts, final checkpoint) are to be stored.
+- `$ZONE`: Compute zone (e.g. `central1-a`).
+
+Optional environment variables:
+
+- `$REPO`: Alternative repo to use instead of the default
+  https://github.com/google/flax - this is useful for development.
+- `$BRANCH`: Alternative branch to use instead of the default `master`.
```
Comment (avital) [inline linen_examples/cloud/README.md:48]: This is great!

File: linen_examples/cloud/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  --repo=${REPO:-https://github.com/google/flax} \
+  --branch=${BRANCH:-master} \
+  --example=mnist \
+  --args='--config=configs/default.py' \
+  --name=default
+```
+
+## Training the imagenet example
+
+Note that you need to first prepare `imagenet2012` dataset. For this, download
```
Comment (avital) [inline linen_examples/cloud/README.md:70]: prepare the `imagenet2021` dataset

File: linen_examples/cloud/README.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+the data from http://image-net.org/ as described in the
+[tensorflow_datasets catalog](https://www.tensorflow.org/datasets/catalog/imagenet2012)
+and then prepare the dataset by running
+
+```shell
+python -c "
+import tensorflow_datasets as tfds
+tfds.builder('imagenet2012').download_and_prepare(
+    download_config=tfds.download.DownloadConfig(
+        manual_dir='$IMAGENET_DOWNLOAD_PATH'))
```
Comment (avital) [inline linen_examples/cloud/README.md:80]: Is the user supposed to set $IMAGENET_DOWNLOAD_PATH? That's not specified anywhere, it seems?

Comment (avital) [reply linen_examples/cloud/README.md:80]: Or is it actually `.` (current directory)?

File: linen_examples/cloud/launch_gce.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    False,
+    help='If set, then the command to launch the GCE instance will only be '
+    'printed to stdout.')
+
+# Machine configuration.
+flags.DEFINE_string('project', None, help='Name of the Google Cloud project.')
+flags.DEFINE_string('zone', None, help='Zone in which the VM will be created.')
+flags.DEFINE_string(
+    'machine_type', None, help='Zone in which the VM will be created.')
+flags.DEFINE_string(
```
Comment (avital) [inline linen_examples/cloud/launch_gce.py:25]: Maybe add examples for what this flag can be? Like, is it `"v100"`?

File: linen_examples/cloud/launch_gce.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+      '--boot-disk-type=pd-ssd', '--metadata=install-nvidia-driver=True',
+      f'--metadata-from-file=startup-script={startup_script}'
+  ]
+  if FLAGS.gpu_type and FLAGS.gpu_count:
+    args.extend([
+        '--maintenance-policy=TERMINATE',
+        f'--accelerator=type=nvidia-tesla-{FLAGS.gpu_type},count={FLAGS.gpu_count}',
+    ])
+
+  if FLAGS.dry_run:
```
Comment (avital) [inline linen_examples/cloud/launch_gce.py:110]: It's very cool to have a dry run option.

**Review by avital — APPROVED** [review]
I love this! And the way it guides people who aren't familiar with tmux...

File: linen_examples/cloud/README.md (hunk 3 above)
Comment (andsteing) [reply linen_examples/cloud/README.md:70]: Added "the".

File: linen_examples/cloud/launch_gce.py (hunk 5 above)
Comment (andsteing) [reply linen_examples/cloud/launch_gce.py:25]: Good point. Added command to list these.

File: linen_examples/cloud/README.md (hunk 4 above)
Comment (andsteing) [reply linen_examples/cloud/README.md:80]: Explained this better.

**Pull request merged**